neofetch

```diff
@@ -3202,6 +3202,49 @@ END
             # Default fallback font hardcoded in terminal-preferences.c
             [[ -z "$term_font" ]] && term_font="Monospace 12"
         ;;
+
+        "conemu-"*)
+            local ce_arg_list
+            local ce_arg_idx
+            local ce_conf
+
+            # Could have used `eval set -- "$ConEmuArgs"` instead for arg parsing but eval is evil...
+            readarray -t ce_arg_list < <(xargs -n1 printf "%s\n" <<< "${ConEmuArgs-}")
+
+            for ce_arg_idx in "${!ce_arg_list[@]}"; do
+                # Search for "-LoadCfgFile" arg
+                [[ "${ce_arg_list[$ce_arg_idx]}" != -LoadCfgFile ]] && continue
+
+                # Conf path is the next arg
+                ((++ce_arg_idx))
+                ce_conf="${ce_arg_list[$ce_arg_idx]}"
```

Comment:
```sh
ce_conf=${ce_arg_list[++ce_arg_idx]}
```

Can then remove the above line too.

Reply:
Uh, I didn't know you could write this. I found `${ce_arg_list[$((++ce_arg_idx))]}` too cumbersome. Nice to know, thanks !

Reply:
Yeah, all arithmetic works inside `${arr[]}`!

Even stuff like this:

```sh
var=${arr[i > j ? i : j]}
```

Reply:
You can do this:

```sh
read -ra ce_arg_list <<< "$ConEmuArgs"

for ce_arg_idx in "${!ce_arg_list[@]}"; do
    # Search for "-LoadCfgFile" arg.
    [[ "${ce_arg_list[$ce_arg_idx]}" != -LoadCfgFile ]] && continue

    # Conf path is the next arg.
    ce_conf=${ce_arg_list[++ce_arg_idx]}
    break
done
```

Reply:
I think this is even simpler though:

```sh
[[ "$ConEmuArgs" =~ -LoadCfgFile\ +([^ ]+) ]] &&
    ce_conf=${BASH_REMATCH[1]}
```

You can see the REGEX in action here: https://regex101.com/r/IoZVQw/1

This avoids yet another array/local variable as well as the loop entirely!

Reply:
You could even drop the `ce_conf` variable and use `${BASH_REMATCH[1]}` directly.

Reply:
Oops, I put this in the wrong comment thread didn't I?

Reply:
`-LoadCfgFile "C:\Users\Steve\AppData\Roaming\Conemu with spaces.xml"` won't be parsed correctly this way.

Reply:
```sh
[[ "$ConEmuArgs" =~ -LoadCfgFile\ +([^ ]+) ]] &&
    ce_conf=${BASH_REMATCH[1]//[\'\"]}
```

The regex still matches when it is quoted so we can just strip the quotes if they exist.